backend/src/v4/models/chatEvent.js

```diff
@@ -217,34 +223,41 @@ const subscribeToV5Events = () => {
 			}
 		}
 		if(message) {
-			const Mailer = require("../mailer/mailer");
 			const notes = await notifications.insertModelUpdatedFailedNotifications(teamspace, model, user, message);
 			notes.forEach((note) => upsertedNotification(null, note));
 
 			if(!userErr) {
-				const attachments = [];
-				const path = require("path");
-				const sharedSpacePath = require("../config").cn_queue.shared_storage;
-				const sharedDir = path.join(sharedSpacePath, corId);
-				const files = require("fs").readdirSync(sharedDir);
-				files.forEach((file) => {
-					if(file.endsWith(".log")) {
-						attachments.push({
-							filename: file,
-							path: path.join(sharedDir, file)
+				try {
+					const archive = archiver("zip", { zlib: { level: 1 } });
+					const sharedDir = path.join(sharedSpacePath, corId);
+					const files = fs.readdirSync(sharedDir);
+
+					files.forEach((file) => {
+						if (file.endsWith(".log")) {
+							archive.file(path.join(sharedDir, file), { name: file });
+						}
+					});
+
+					const zipPath = path.join(sharedDir, "logs.zip");
+					const output = fs.createWriteStream(zipPath);
+
+					output.on("close", () => {
```

Comment:
As it stands it's a callback function so the this function can easily exits before the callback is called. If sendImportError errors out, ti's entering no mans land and we are probably going to crash the server again with unresolved promise rejection.

So I think we need to create a promise that is only resolved in the callback function and wait on it would be a better managed flow.

Reply:
It would also help to tap on the errors and reject the promise (same with the archiver stuff)